configure.ac

```diff
@@ -483,6 +485,45 @@ AC_CHECK_HEADERS([sys/types.h signal.h stddef.h], [],
 # optional:
 AC_CHECK_HEADERS([sys/time.h locale.h fcntl.h dlfcn.h sys/wait.h sys/sysmacros.h])
 
+# Check for JNI
+AC_ARG_ENABLE([java],
+              [AC_HELP_STRING([--disable-java],
+                              [disable Java Interoperability])])
+AS_IF([test X$enable_java != Xno], [
```

Comment:
`ARC_ARG_ENABLE` is more about optional features (like the debug log) and less about external packages where `AC_ARG_WITH` is preferable (and would also match what configure has for example for libxml2).

See https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.72/html_node/External-Software.html and adjust (you may postpone that part until you have actual running tests, though).

In any case please move the JNI block directly before the libxml2 part.